.NET/Microsoft.Recognizers.Text.NumberWithUnit/Parsers/NumberWithUnitParser.cs

```diff
@@ -136,6 +136,24 @@ public ParseResult Parse(ExtractResult extResult)
                         resolutionStr += halfValue?.ResolutionStr.Substring(1);
                     }
 
+                    // In certain cultures the unit can be split around the number,
+                    // e.g. in Japanese "秒速100メートル" ('speed per second 100 meters' = 100m/s).
+                    // Here prefix and suffix are combined in order to parse the unit correctly.
+                    if (unitValue == Constants.PARTIAL_UNIT && unitKeys.Count > 1 && this.Config.CheckFirstSuffix)

```

Comment:
Don't other cultures already have a mechanism for this?

Reply:
yes, but here it is a bit different. For common split units encountered so far, one part is sufficient to determine the whole unit, e.g. "Degree 30 Celsius". In this case ('speed per second 100 meters') both parts are required to parse the unit correctly.